src/waveform/renderers/waveformrenderbeat.cpp

```diff
@@ -83,11 +87,29 @@ void WaveformRenderBeat::draw(QPainter* painter, QPaintEvent* /*event*/) {
 
         xBeatPoint = qRound(xBeatPoint);
 
+        int beatIndex = it->beatIndex();
+
         // If we don't have enough space, double the size.
         if (beatCount >= m_beats.size()) {
             m_beats.resize(m_beats.size() * 2);
         }
 
+        if(beatIndex >= 0) {
```

Comment:
This logic needs to be moved at least to the 
Iterator. 
I suggest a isBar() and isPhraseStart()
A phrase does not necessarily strat with a bar.